### PR TITLE
Adding row/column tests and benchmarks along with a benchmark/test building framework

### PR DESCRIPTION
**Pull request by hyperbolic2346**
This PR adds c++ tests and benchmarks to `spark-rapids-jni`. The tests and benchmarks link in the helpers from cudf, so a test can be lifted directly from cudf or written in the same style as a cudf test. The benchmarks only support nvbench, as that is the preferred benchmarking tool.

To build with tests, pass `-DBUILD_TESTS` to the command line for the build and to build benchmarks, pass `-DBUILD_BENCHMARKS`. The test library used from cudf is built any time cudf is built, so no changes are necessary for that build. The benchmark library requires that benchmarks are built, so this is plumbed down through the `pom.xml` and `build-libcudf.xml` if the user selects benchmark building.

The change includes a new script `build/run-in-docker` that will run an executable in the same container environment that was used to build `spark-rapids-jni`. This script also allows an empty argument list and will drop the user into an interactive shell inside the container. This isn't a requirement, but ensures the same environment in the case the user is unable or unwilling to match the build environment.

I had to copy the 22.08 version of `row_conversion.cu` so the tests would properly work, but it is not quite ready to move to this repository yet.

Adding automatic running of tests after build was suggested as a feature, but left for another PR. I have added issue #330 to track it.

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
The contributing guide needs to be updated with instructions on how to use the new build features.

File: build-libcudf.xml [hunk 1]
```diff
@@ -38,6 +38,7 @@
       <arg value="${cudf.path}/cpp"/>
       <arg value="-DBUILD_SHARED_LIBS=OFF"/>
       <arg value="-DBUILD_TESTS=OFF"/>
+      <arg value="-DBUILD_BENCHMARKS=${BUILD_BENCHMARKS}"/>
```
[the remark this answers is not in the document]
Comment (hyperbolic2346) [reply build-libcudf.xml:41]: The goal isn't to run cudf benchmarks or their tests. We do want to leverage their libraries for data generation, test printing data frames, etc. For tests this library is public and built even when tests are not, but it isn't public or built for the benchmarks, so we have to build their benchmarks so we can link against that library. Without it, we would be reproducing content like [generate_input.cu](https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/benchmarks/common/generate_input.cu).

[the remark this answers is not in the document]
Comment (hyperbolic2346) [reply build-libcudf.xml:41]: I should say that I asked if the benchmark library could be made public and was told that wasn't desired or likely to happen.

[the remark this answers is not in the document]
Comment (jlowe) [reply build-libcudf.xml:41]: Ah, would be good to add a comment explaining this.  Might also be worth filing a feature request where we can build this without building all benchmarks (which last checked took quite some time).

**Review by gerashegalov — COMMENTED** [review]
LGTM

**Review by gerashegalov — APPROVED** [review]

File: build-libcudf.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hyperbolic2346) [reply build-libcudf.xml:44]: Added a comment. Pulling the common benchmark files over turned out to be non-trivial, so I'm going to add an issue to track that and leave this for now.

**Review by jlowe — DISMISSED** [review]

**Comment by hyperbolic2346** [comment]
Yes, this turned out to be a bug in the CMake Makefile Generator. I have a workaround and will update with it soon.

**Comment by hyperbolic2346** [comment]
build

**Comment by hyperbolic2346** [comment]
The workaround isn't working so well. Talking with Robert Maynard about it.

**Comment by hyperbolic2346** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**